### PR TITLE
Respect nullable kwarg for TimestampAttribute

### PR DESCRIPTION
**Pull request by th0114nd**

### CODE REVIEW

File: pynamodb_attributes/timestamp.py [hunk 1]
```diff
@@ -25,10 +25,14 @@ def serialize(self, value: datetime) -> str:
         return str(int(value.timestamp() * self._multiplier))
 
     def __set__(self, instance: Any, value: Optional[Any]) -> None:
```
Comment (ikonst) [inline pynamodb_attributes/timestamp.py:27]: PynamoDB already validates nulls during serialization. If we want it to happen earlier (on assignment), we should probably change `Attribute.__set__` rather than do it in every attribute.

Can we instead wrap with `if value is not None:`, e.g.
```python
if value is not None:
    if not isinstance(value, datetime):
        raise TypeError(f"value has invalid type '{type(value)}'; datetime expected")
    if value.tzinfo is None or value.tzinfo.utcoffset(value) is None:
        raise TypeError("aware datetime expected")

return super().__set__(instance, value)
```

**Review by ikonst — CHANGES_REQUESTED** [review]
See comment.

File: pynamodb_attributes/unicode_delimited_tuple.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def deserialize(self, value: str) -> T:
     def serialize(self, value: T) -> str:
         if not isinstance(value, self.tuple_type):
             raise TypeError(f"value has invalid type '{type(value)}'; expected '{self.tuple_type}'")
-        values = [e for e in value]
+        values = list(value)
```
Comment (ikonst) [inline pynamodb_attributes/unicode_delimited_tuple.py:54]: unrelated change? (though I agree in spirit)

**Review by ikonst — APPROVED** [review]

File: pynamodb_attributes/unicode_delimited_tuple.py (hunk 2 above)
Comment (th0114nd) [reply pynamodb_attributes/unicode_delimited_tuple.py:54]: Yes, but Travis gave me a red x for “ '[C416] Unnecessary list comprehension'”. Not sure if the linter added a rule recently or what

Comment (mxr) [reply pynamodb_attributes/unicode_delimited_tuple.py:54]: Since `flake8-comprehensions` is unpinned this kind of thing may happen from time to time. We could pin it
